### ISSUE TITLE
Error calling astype on singletons

### ISSUE THREAD

**Issue by freeman-lab** [thunder-project/thunder#121]
Calling astype on a `Data` object with singleton entries yields an error. To reproduce, use:

```
data = tsc.loadExample('fish-series')
data.seriesMean().astype('int').first()
```
The error is `TypeError: astype() takes no keyword arguments`. It appears that when calling `astype` on a `numpy.float64` (or other scalar), it doesn't take the same any additional arguments that work when calling it on an array. For example,

```
x = np.array([1.0,2.0,3.0])
x.astype('int', casting='unsafe')
```
works fine but this doesn't
```
x = np.float64(1.0)
x.astype('int', casting='unsafe')
```
One potential solution would be to check the length of the index before calling astype, and to only include the additional arguments if it's greater than 1.

**Comment by industrial-sloth** [comment]
Well that's irritating. I can confirm, at least with numpy 1.8.2 (which is what I happen to have installed here), numpy scalars don't appear to support keyword arguments in `astype` in the same way that numpy arrays do. Yeah, we'll add some type checking here - too bad, otherwise this was a pretty simple function...

**Comment by eric-wieser** [comment]
This will be fixed in numpy 1.14
